controllers/metal3machine_controller.go

```diff
@@ -278,7 +330,7 @@ func (r *Metal3MachineReconciler) Metal3ClusterToMetal3Machines(o handler.MapObj
 		if m.Spec.InfrastructureRef.Name == "" {
 			continue
 		}
-		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
+		name := client.ObjectKey{Namespace: m.Spec.InfrastructureRef.Namespace, Name: m.Name}
```

Comment:
This might be empty, you need to default it to m.Namespace if m.Spec.InfrastructureRef.Namespace == ""

Reply:
empty-case is checked in line 330. I don't that what you suggested is necessary

Reply:
Name and namespace are different. You might have empty namespace but given name

Reply:
Is it possible? Didn't know that? OK, I will default it